contrib/kubespray/roles/requirement/worker/tasks/ubuntu.yml

```diff
@@ -0,0 +1,94 @@
+---
+- name: "Worker-Task-1-[1/2] Check whether the vm can access to docker.io"
+  raw: nslookup index.docker.io
+  register: worker_nslookup_docker_io
+  failed_when: false
+  changed_when: false
+  check_mode: false
+  environment: {}
+
+- name: "Worker-Task-1-[2/2] Check whether the vm can access to docker.io"
+  fail:
+    msg: "Unable to access docker.io"
+  changed_when: false
+  check_mode: false
+  environment: {}
+  when:
+    - worker_nslookup_docker_io.rc != 0
+
+- name: "Worker-Task-2-[1/3] Check whether the NTP is installed and enabled."
+  raw: pgrep ntpd
+  register: worker_pgrep_ntpd
+  failed_when: false
+  changed_when: false
+  check_mode: false
+  environment: {}
+
+- name: "Worker-Task-2-[2/3] Check whether the NTP is installed and enabled."
+  raw: ntpq -p
+  register: worker_ntpq
+  failed_when: false
+  changed_when: false
+  check_mode: false
+  environment: {}
+
+- name: "Worker-Task-2-[3/3] Check whether the NTP is installed and enabled."
+  fail:
+    msg: "NTP is not enabled on your worker machines."
+  changed_when: false
+  check_mode: false
+  environment: {}
+  when:
+    - worker_ntpq.rc != 0 or worker_nslookup_docker_io.rc != 0
+
+- name: "Worker-Task-3-[1/2] Check GPU card info"
+  raw: "lspci | grep -E '[0-9a-fA-F][0-9a-fA-F]:[0-9a-fA-F][0-9a-fA-F].[0-9] (3D|VGA compatible) controller: NVIDIA Corporation.*'"
+  register: worker_lspci_grep
+  failed_when: false
+  changed_when: false
+  check_mode: false
+  environment: {}
+
+- name: "Worker-Task-3-[2/2] Check GPU card info"
+  fail:
+    msg: "NVIDIA GPU card is not detected in your worker machines."
+  changed_when: false
+  check_mode: false
+  environment: {}
+  when:
+    - worker_lspci_grep.rc != 0
+
+- name: "Worker-Task-4-[1/2] Check NVIDIA GPU drivers"
+  raw: "nvidia-smi"
+  register: worker_nvidia_smi
+  failed_when: false
+  changed_when: false
+  check_mode: false
+  environment: {}
+
+- name: "Worker-Task-4-[2/2] Check NVIDIA GPU drivers "
+  fail:
+    msg: "NVIDIA GPU drivers is not detected in your worker machines."
+  changed_when: false
+  check_mode: false
+  environment: {}
+  when:
+    - worker_nvidia_smi.rc != 0
+
+- name: "Worker-Task-5-[1/2] Check NVIDIA container runtime"
+  raw: "command -v nvidia-container-runtime "
+  register: worker_nvidia_container_runtime
+  failed_when: false
+  changed_when: false
+  check_mode: false
+  environment: {}
+
+- name: "Worker-Task-5-[2/2] Check NVIDIA container runtime"
+  fail:
+    msg: "NVIDIA container runtime is not detected in your worker machines."
+  changed_when: false
+  check_mode: false
+  environment: {}
+  when:
+    - worker_nvidia_container_runtime.rc != 0
+
```

Comment:
should check whether nvidia is the default runtime:
```sh
[ "$(docker info -f '{{json .DefaultRuntime}}')" = "\"nvidia\"" ]
```

Reply:
Docker environment  and installation will be covered by kubespray.

Reply:
Default runtime is set here: https://github.com/microsoft/pai/blob/master/contrib/kubespray/quick-start-kubespray.sh#L51